### ISSUE TITLE
AWX running on Kubernetes is unable to handle instances capacity when running many jobs simultaniously.

### ISSUE THREAD

**Issue by cesan3** [ansible/awx#1873]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
We schedule a run of a playbook on all our inventories. We expect the instances handle the load correctly, but we see weird behaviours when playbooks are run against big inventories.

##### ENVIRONMENT
* AWX version: 1.0.6.5
* AWX install method: kubernetes
* Ansible version:  2.5.2
* Operating System: Linux
* Web Browser: Firefox/Chrome

##### STEPS TO REPRODUCE

Have a small playbook that checks the expiration date of an SSL certificate.
Have 100 Inventories, varying from 2 hosts, to 150 hosts.
Have a kubernetes cluster with 5 nodes (4 cpus / 12G ram )
Deploy awx and scale it to 5 instances
Set the instances capacity to 70% 
Create an instance group with 4 instances only (80%)
Associate the playbook to that instance group
Run the playbook against all inventories.

##### EXPECTED RESULTS

The instances handle capacity correctly and all the playbooks complete.

##### ACTUAL RESULTS
Very erratical results, several cases.

1.  After capacity is reached and playbooks seemingly run smoothly, one instance disappears, I guess by the AutoScaler
2.  That causes chaos, and capacity never gets distributed among the nodes, only 1 node seem to handle everything. It seems all the jobs that were kicked out of the dropped instances(s), end-up as part of the capacity calculacion for the remaining node , showin percentages going over +100%, 550% while I was writing this.

##### ADDITIONAL INFORMATION

**Comment by jmighion** [comment]
Sounds similar to #701

**Comment by chrismeyersfsu** [comment]
@cesan3 From your experimental setup description; you don't have any jobs running in parallel. Which leads me to believe you have a singular large job (presumably a high fork count set). AWX always allows at least 1 job to run (even it if will blow out the capacity). My guess is that you are OOM and killing nodes. By default, AWX will ask for 3 compute units 6GB of ram. You will want to raise that resource request.

https://github.com/ansible/awx/blob/ffc347bfd0be0ea166ab5340a6ced0f4d16dd39e/installer/roles/kubernetes/defaults/main.yml
https://github.com/ansible/awx/blob/devel/installer/roles/kubernetes/templates/deployment.yml.j2#L143

**Comment by jmighion** [comment]
@chrismeyersfsu 
```
Have 100 Inventories, varying from 2 hosts, to 150 hosts.
Run the playbook against all inventories
```
Seems like 100 jobs in parallel.

**Comment by chrismeyersfsu** [comment]
To have those jobs run in parallel you would need to either (1) create a playbook for each inventory or (2) set allow_simultaneous to True on the Job Template + turn on prompt for inv on launch. I did not see those steps in the steps to reproduce.

**Comment by cesan3** [comment]
To give more details on my setup and to clarify it a bit more for @chrismeyersfsu, I have set up those in parallel:
My template has the allow_simoultaneous flag to True, and I prompt for inv, credentials, extra vars and so on.
The way I launch it is by using tower-cli in a for loop:
for client in `cat /tmp/all-clients`; do tower-cli job launch -J <id> --inventory $client --credential $client --no-input -e "special_var=val"; sleep 2; done
This will execute the jobs and insert them in the queue, which will be processed after.
After the capacity is reached, and jobs continue getting added to the queue, is where I start to see weird behaviour, depending on the playbook.

**Comment by cesan3** [comment]
@chrismeyersfsu , Additionally, I've tried to mitigate this by keeping the capacity of each instance at 70% marker. I've also, explicitly associate some instances to each instance group, with no difference.

**Comment by chrismeyersfsu** [comment]
@cesan3 we've merged in many fixes to the task manager in the new 1.0.7 AWX release. Can you give your experiment another try?

**Comment by cesan3** [comment]
Hello @chrismeyersfsu.  Capacity is correctly handled in version 1.0.7.0.  The cluster is very stable now.
Thanks.
